src/ngraph/pass/reshape_sinking.cpp

```diff
@@ -175,6 +175,20 @@ void swim(Input<Node> input, shared_ptr<op::Reshape> reshape)
         auto csw = work_queue.front();
         work_queue.pop_front();
         auto n = csw.input.get_source_output().get_node_shared_ptr();
+
+        auto materialize = [csw, n]() {
+            auto new_reshape = csw.reshape->copy_with_new_args({n});
+            new_reshape->merge_provenance_tags_from(n);
+            NGRAPH_DEBUG << "Materializing new reshape " << describe_reshape(new_reshape);
+            csw.input.replace_source_output(new_reshape->output(0));
+        };
+
+        // Only swim past nodes which have a single user
+        if (n->get_users(true).size() > 1)
```

Comment:
would this get_users(true) will increase the compile time? 
I remember we had an issue sometime back on slice reshape sinking, this caused significant time increase on reshape sinking, so we had to change to just get_users().

Reply:
It could if there are many ops which have long dangling output chains that never make it to a result -- since get_users(true) is making sure the users it counts are used by Results (calling ngraph::is_used).

I suspect that is unlikely but if you have evidence in the past to the contrary then it's worth checking the compile time.

Reply:
We will run all CI tests first with all our models to make sure that no regression. Will update.